plugins/outputs/graylog/graylog.go

```diff
@@ -339,36 +346,118 @@ func (g *Graylog) Connect() error {
 		return err
 	}
 
-	for _, server := range g.Servers {
-		w := newGelfWriter(gelfConfig{Endpoint: server}, dialer, tlsCfg)
-		err := w.Connect()
-		if err != nil {
-			return fmt.Errorf("failed to connect to server [%s]: %v", server, err)
-		}
-		writers = append(writers, w)
-		g.closers = append(g.closers, w)
+	if g.Reconnection {
+		go g.connectRetry(tlsCfg)
+		return nil
 	}
 
+	unconnected, gelfs := g.connectEndpoints(g.Servers, tlsCfg)
+	if len(unconnected) > 0 {
+		servers := strings.Join(unconnected, ",")
+		return fmt.Errorf("connect: connection refused for %s", servers)
```

Comment:
"connection refused" means something somewhat specific, while the failure could have been for something else.

How about something like "connection failed for %s"?